docs/metrics/average-precision.md

```diff
@@ -0,0 +1,250 @@
+# Average Precision
+
+Average precision (AP) summarizes a precision-recall (PR) curve into a single value representing the average of all
+precisions. It is generally understood as the approximation of the area under the PR curve. This metric ranges between
+0 to 1, where a perfect model with a [precision](./precision.md) and [recall](./recall.md) of 1 scores 1.
+The larger the metric, the better a model performs across different thresholds.
+
+!!! info inline end "Guides: Precision and Recall"
+
+    Read the [precision](./precision.md) and the [recall](./recall.md) guides if you're not familiar with those metrics.
+
+Unlike metrics like [precision](./precision.md), [recall](./recall.md), and [F<sub>1</sub>-score](./f1-score.md), which
+are threshold-dependent where a confidence threshold value must be defined to compute them, AP is a key performance
+metric that removes the dependency of selecting one confidence threshold value and measures a model's performance across
+all thresholds.
+
+AP is commonly used to evaluate the performance of object detection and information retrieval workflows. This metric
+(or an aggregated version of it called **mean average precision** (mAP)) is the primary metric used across popular
```

Comment:
We should add a subsection for mAP — just a simple description like this, along with something saying that it's usually the macro-average across all classes.